### PR TITLE
M3-2108: Change: Refactor domains dashboard card to use Redux state

### PR DESCRIPTION
**Pull request by johnwcallahan**
## Description

Refactored DomainsDashboardCard to make use of Redux State. This refactor was based off of LinodesDashboardCard. The driver for this was the fact there was an unnecessary request to `/domains` happening on the dashboard.

## Type of Change
- Non breaking change ('update', 'change')

## Note to Reviewers
 To verify:

**On production:**
-  Go to the Dashboard.
-  Open up Network dev tools, and you'll notice two separate GET requests to /domains:

screen shot 2018-12-28 at 5 37 17 pm

**On this branch:**

- Take the same steps as above, and you'll notice ONE request now:
screen shot 2018-12-28 at 5 37 03 pm

### CODE REVIEW

File: src/features/Dashboard/DomainsDashboardCard/DomainsDashboardCard.tsx [hunk 1]
```diff
@@ -167,8 +119,53 @@ class DomainsDashboardCard extends React.Component<CombinedProps, State> {
...
+      sortBy(prop('domain')),
+    )(state.__resources.domains.entities),
+    loading: state.__resources.domains.loading,
+    error: state.__resources.domains.error,
+  };
+});
+
+const mergeEvents = (events: Linode.Event[]) => (domains: Linode.Domain[]) =>
+  events
+    .reduce((updatedLinodes, event) => {
```
Comment (Jskobos) [inline src/features/Dashboard/DomainsDashboardCard/DomainsDashboardCard.tsx]: Domains

File: src/features/Dashboard/DomainsDashboardCard/DomainsDashboardCard.tsx [hunk 2]
```diff
@@ -167,8 +119,53 @@ class DomainsDashboardCard extends React.Component<CombinedProps, State> {
...
+      if (isWantedEvent(event)) {
+        return updatedLinodes.map(domain => event.entity.id === domain.id
```
Comment (Jskobos) [inline src/features/Dashboard/DomainsDashboardCard/DomainsDashboardCard.tsx]: Domains

File: src/features/Dashboard/DomainsDashboardCard/DomainsDashboardCard.tsx [hunk 3]
```diff
@@ -167,8 +119,53 @@ class DomainsDashboardCard extends React.Component<CombinedProps, State> {
...
+          ? { ...domain, recentEvent: event }
+          : domain
+        )
+      }
+
+      return updatedLinodes;
```
Comment (Jskobos) [inline src/features/Dashboard/DomainsDashboardCard/DomainsDashboardCard.tsx]: Domains

File: src/features/Dashboard/DomainsDashboardCard/DomainsDashboardCard.tsx (hunk 1 above)
Comment (rjstires) [reply src/features/Dashboard/DomainsDashboardCard/DomainsDashboardCard.tsx]: `updateThings`

**Pull request merged**